modules/cnf-about-topology-aware-lifecycle-operator-config.adoc

```diff
@@ -0,0 +1,127 @@
++
+[NOTE]
+====
+You can only make changes to the `spec` fields if the `ClusterGroupUpgrade` CR is either in the `UpgradeNotStarted` or the `UpgradeCannotStart` state. 
+====
++
+.Sample CR
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+  namespace: default
+spec:
+  clusters: <1>
+  - spoke1
+  - spoke2
+  - spoke3
+  enable: false
+  managedPolicies:
+  - policy1-common-cluster-version-policy
+  - policy2-common-pao-sub-policy
+  - policy3-common-ptp-sub-policy
+  remediationStrategy: <2>
+    canaries: <3>
+    - spoke1
+    maxConcurrency: 2 <4>
+    timeout: 240
+status: <5>
+  conditions:
+  - message: The ClusterGroupUpgrade CR is not enabled
+    reason: UpgradeNotStarted
+    status: "False"
+    type: Ready
+----
+<1> The list of clusters to update.
+<2> The `remediationStrategy` fields define the specifics of the cluster updates.
+<3> The remediation plan starts with the clusters listed in the `canaries` field. Each canary cluster forms a single-cluster batch.
+<4> The `maxConcurrency` field defines the maximum number of concurrent updates in a batch. The number of remediation batches is the number of canary clusters, plus the number of clusters (except the canary clusters) divided by `maxConcurrency`. The clusters that are already `Compliant`` with all the managed policies are excluded from the remediation plan.
+<5> The `status` field displays information about the status of the updates.
+
+`UpgradeNotCompleted`:: In this state, the Operator enforces the policies following the remediation plan defined in the `UpgradeNotStarted` state.
++
+Enforcing the policies of subsequent batches starts immediately after all the clusters of the current batch are `Compliant` with all the managed policies. If the batch times out, the Operator moves on to the next batch. The timeout value of a batch  is the `spec.timeout` field divided by the number of batches in the remediation plan.
++
+[NOTE]
+====
+The managed policies are applied in the order they are listed in the `managedPolicies` field in the `ClusterGroupUpgrade` CR.
+====
++
+The Operator transitions to the `UpgradeTimedOut` state in two cases:
++
+* If the current batch is a canary type and the cluster in the batch does not change to `Compliant` with all the managed policies
+* If the clusters are not `Compliant` with the managed policies within the `timeout` value specified in `remediationStrategy`
+
+`UpgradeTimedOut`:: In this state, the Operator periodically checks if all the policies for the `ClusterGroupUpgrade` are compliant.
+The periodic checks allow the updates to complete if the updates are prolonged due to network, CPU, or other issues. If the policies are compliant, the Operator transitions to the `UpgradeCompleted` state.
```

Comment:
@irinamihai Could you take a look at Ian's question?

Reply:
Yes, every one hour until the CGU is deleted/completed, per the initial design.

Reply:
Added that detail:
> `UpgradeTimedOut`:: In this state, the Operator checks every hour if all the policies for the `ClusterGroupUpgrade` CR are compliant. The periodic checks continue until the update is completed or the `ClusterGroupUpgrade` CR is deleted.

Reply:
Yes. This was added as it was agreed during design that we didn't want a sink/failed state, but rather a timed out state that would allow the controller to keep getting updates from policies in case they completed after the window.